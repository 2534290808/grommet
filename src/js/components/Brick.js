// (C) Copyright 2014-2015 Hewlett Packard Enterprise Development LP

import React, { Component } from 'react';
import Anchor from './Anchor';

const CLASS_ROOT = "brick";

export default class Brick extends Component {
  render() {
    var classes = [CLASS_ROOT];
    classes.push('brick--' + this.props.width + '-' + this.props.height);

    if (this.props.colorIndex) {
      classes.push('background-color-index-' + this.props.colorIndex);
    }
    if (this.props.className) {
      classes.push(this.props.className);
    }

    var label = <div className="brick--label"><span>{this.props.label}</span></div>;
    if (this.props.route) {
      label = <Anchor href={this.props.route} className="brick--label"><span>{this.props.label}</span></Anchor>;
    }

    return (
      <div className={classes.join(' ')} onClick={this.props.onClick}>
        <div className="brick--content-wrapper">
          {this.props.children}
        </div>
        {label}
      </div>
    );
  }
}

Brick.propTypes = {
  label: React.PropTypes.string,
  route: React.PropTypes.string,
  width: React.PropTypes.number,
  height: React.PropTypes.number,
  colorIndex: React.PropTypes.string
};

Brick.defaultProps = {
  width: 1,
  height: 1
};
