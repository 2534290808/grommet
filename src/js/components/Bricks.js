// (C) Copyright 2014-2015 Hewlett Packard Enterprise Development LP

import React, { Component } from 'react';
import Box from './Box';

const CLASS_ROOT = "bricks";

export default class Bricks extends Component {
  render () {
    var classes = [CLASS_ROOT];

    if (this.props.className) {
      classes.push(this.props.className);
    }

    return (
      <Box ref="bricks" className={classes.join(' ')}>
        {this.props.children}
      </Box>
    );
  }
}
