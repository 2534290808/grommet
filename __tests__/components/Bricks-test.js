// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import Brick from '../../src/js/components/Brick';
import Bricks from '../../src/js/components/Bricks';
import FakeIcon from '../mocks/FakeIcon';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('Bricks', () => {
  it('renders with the correct default options', () => {
    const component = renderer.create(
      <Bricks>
        <Brick label="First" />
        <Brick label="Second" />
        <Brick label="Third" />
        <Brick label="Fourth" />
      </Bricks>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly when setting the type of Brick children', () => {
    const component = renderer.create(
      <Bricks>
        <Brick label="First" type="small" />
        <Brick label="Second" type="large" />
        <Brick label="Third" type="wide" />
        <Brick label="Fourth" type="tall" />
      </Bricks>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly when setting the colorIndex of Brick children', () => {
    const component = renderer.create(
      <Bricks>
        <Brick label="First" colorIndex="neutral-1" />
        <Brick label="Second" colorIndex="neutral-2" />
        <Brick label="Third" colorIndex="neutral-3" />
        <Brick label="Fourth" colorIndex="neutral-4" />
      </Bricks>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders a complex example with multiple custon props', () => {
    const component = renderer.create(
      <Bricks>
        <Brick colorIndex="neutral-1" type="large"
          href="http://www.grommet.io/" texture="/img/carousel-1.png" />
        <Brick label="Second" colorIndex="neutral-2" type="tall" />
        <Brick label="Third" colorIndex="neutral-3">
          <FakeIcon />
        </Brick>
        <Brick label="Fourth" colorIndex="neutral-4" />
      </Bricks>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
