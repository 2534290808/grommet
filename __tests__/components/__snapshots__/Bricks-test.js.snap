exports[`Bricks renders a complex example with multiple custon props 1`] = `
<div
  className="">
  <a
    aria-label={undefined}
    className="grommetux-anchor grommetux-brick grommetux-brick--2-2 grommetux-background-color-index-neutral-1 grommetux-brick--clickable"
    href="http://www.grommet.io/"
    id={undefined}
    onClick={undefined}
    target={undefined}>
    <div
      className="grommetux-brick__background"
      style={
        Object {
          "background": "url(/img/carousel-1.png) no-repeat center center",
          "backgroundSize": "cover"
        }
      }>
      <div>
        <div
          className="grommetux-brick__container" />
        <div
          className="grommetux-brick__label">
          <span />
        </div>
      </div>
    </div>
  </a>
  <div
    className="grommetux-brick grommetux-brick--1-2 grommetux-background-color-index-neutral-2"
    style={Object {}}>
    <div>
      <div
        className="grommetux-brick__container" />
      <div
        className="grommetux-brick__label">
        <span>
          Second
        </span>
      </div>
    </div>
  </div>
  <div
    className="grommetux-brick grommetux-brick--1-1 grommetux-background-color-index-neutral-3"
    style={Object {}}>
    <div>
      <div
        className="grommetux-brick__container">
        <svg
          height="24px"
          role="img"
          version="1.1"
          viewBox="0 0 24 24.3616"
          width="24px" />
      </div>
      <div
        className="grommetux-brick__label">
        <span>
          Third
        </span>
      </div>
    </div>
  </div>
  <div
    className="grommetux-brick grommetux-brick--1-1 grommetux-background-color-index-neutral-4"
    style={Object {}}>
    <div>
      <div
        className="grommetux-brick__container" />
      <div
        className="grommetux-brick__label">
        <span>
          Fourth
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Bricks renders correctly when setting the colorIndex of Brick children 1`] = `
<div
  className="">
  <div
    className="grommetux-brick grommetux-brick--1-1 grommetux-background-color-index-neutral-1"
    style={Object {}}>
    <div>
      <div
        className="grommetux-brick__container" />
      <div
        className="grommetux-brick__label">
        <span>
          First
        </span>
      </div>
    </div>
  </div>
  <div
    className="grommetux-brick grommetux-brick--1-1 grommetux-background-color-index-neutral-2"
    style={Object {}}>
    <div>
      <div
        className="grommetux-brick__container" />
      <div
        className="grommetux-brick__label">
        <span>
          Second
        </span>
      </div>
    </div>
  </div>
  <div
    className="grommetux-brick grommetux-brick--1-1 grommetux-background-color-index-neutral-3"
    style={Object {}}>
    <div>
      <div
        className="grommetux-brick__container" />
      <div
        className="grommetux-brick__label">
        <span>
          Third
        </span>
      </div>
    </div>
  </div>
  <div
    className="grommetux-brick grommetux-brick--1-1 grommetux-background-color-index-neutral-4"
    style={Object {}}>
    <div>
      <div
        className="grommetux-brick__container" />
      <div
        className="grommetux-brick__label">
        <span>
          Fourth
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Bricks renders correctly when setting the type of Brick children 1`] = `
<div
  className="">
  <div
    className="grommetux-brick grommetux-brick--1-1"
    style={Object {}}>
    <div>
      <div
        className="grommetux-brick__container" />
      <div
        className="grommetux-brick__label">
        <span>
          First
        </span>
      </div>
    </div>
  </div>
  <div
    className="grommetux-brick grommetux-brick--2-2"
    style={Object {}}>
    <div>
      <div
        className="grommetux-brick__container" />
      <div
        className="grommetux-brick__label">
        <span>
          Second
        </span>
      </div>
    </div>
  </div>
  <div
    className="grommetux-brick grommetux-brick--2-1"
    style={Object {}}>
    <div>
      <div
        className="grommetux-brick__container" />
      <div
        className="grommetux-brick__label">
        <span>
          Third
        </span>
      </div>
    </div>
  </div>
  <div
    className="grommetux-brick grommetux-brick--1-2"
    style={Object {}}>
    <div>
      <div
        className="grommetux-brick__container" />
      <div
        className="grommetux-brick__label">
        <span>
          Fourth
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Bricks renders with the correct default options 1`] = `
<div
  className="">
  <div
    className="grommetux-brick grommetux-brick--1-1"
    style={Object {}}>
    <div>
      <div
        className="grommetux-brick__container" />
      <div
        className="grommetux-brick__label">
        <span>
          First
        </span>
      </div>
    </div>
  </div>
  <div
    className="grommetux-brick grommetux-brick--1-1"
    style={Object {}}>
    <div>
      <div
        className="grommetux-brick__container" />
      <div
        className="grommetux-brick__label">
        <span>
          Second
        </span>
      </div>
    </div>
  </div>
  <div
    className="grommetux-brick grommetux-brick--1-1"
    style={Object {}}>
    <div>
      <div
        className="grommetux-brick__container" />
      <div
        className="grommetux-brick__label">
        <span>
          Third
        </span>
      </div>
    </div>
  </div>
  <div
    className="grommetux-brick grommetux-brick--1-1"
    style={Object {}}>
    <div>
      <div
        className="grommetux-brick__container" />
      <div
        className="grommetux-brick__label">
        <span>
          Fourth
        </span>
      </div>
    </div>
  </div>
</div>
`;
